src/nuclide_header.F90

```diff
@@ -0,0 +1,1156 @@
+        imu   = floor((mu + ONE) / dmu + ONE)
+        ! Adjust so interpolation works on the last bin if necessary
+        if (imu == size(this % scatter, dim=3)) then
+          imu = imu - 1
+        end if
+        f = this % scatter(gout, gin, imu)
+
+      end if
+
+    end function nuclideiso_calc_f
+
+    pure function nuclideangle_calc_f(this, gin, gout, mu, uvw, i_azi, &
+                                          i_pol) result(f)
+      class(NuclideAngle), intent(in) :: this
+      integer, intent(in)              :: gin  ! Incoming Energy Group
+      integer, intent(in)              :: gout ! Outgoing Energy Group
+      real(8), intent(in)              :: mu   ! Angle of interest
+      real(8), intent(in), optional    :: uvw(3) ! Direction vector
+      integer, intent(in), optional    :: i_azi ! Incoming Energy Group
+      integer, intent(in), optional    :: i_pol ! Outgoing Energy Group
+      real(8)                          :: f    ! Return value of f(mu)
+
+      real(8) :: dmu, r
+      integer :: imu
+      integer :: i_azi_, i_pol_
+      if (present(i_azi) .and. present(i_pol)) then
+        i_azi_ = i_azi
+        i_pol_ = i_pol
+      else if (present(uvw)) then
+        call find_angle(this % polar, this % azimuthal, uvw, i_azi_, i_pol_)
+      end if
+
+      if (this % scatt_type == ANGLE_LEGENDRE) then
+        f = evaluate_legendre(this % scatter(gout,gin,:,i_azi_,i_pol_), mu)
+      else if (this % scatt_type == ANGLE_TABULAR) then
+        dmu = TWO / real(this % order - 1)
+        ! Find mu bin algebraically, knowing that the spacing is equal
+        f   = (mu + ONE) / dmu + ONE
+        imu = floor(f)
+        ! But save the amount that mu is past the previous index
+        ! so we can use interpolation later.
+        f = f - real(imu)
+        ! Adjust so interpolation works on the last bin if necessary
+        if (imu == size(this % scatter, dim=3)) then
+          imu = imu - 1
+        end if
+
+        ! Now intepolate to find f(mu)
+        r  = f / dmu
+        f = (ONE - r) * this % scatter(gout,gin,imu,i_azi_,i_pol_) + &
+             r * this % scatter(gout,gin,imu+1,i_azi_,i_pol_)
+      else ! (ANGLE_HISTOGRAM)
+        dmu = TWO / real(this % order)
+        ! Find mu bin algebraically, knowing that the spacing is equal
+        imu   = floor((mu + ONE) / dmu + ONE)
+        ! Adjust so interpolation works on the last bin if necessary
+        if (imu == size(this % scatter, dim=3)) then
+          imu = imu - 1
+        end if
+        f = this % scatter(gout, gin, imu,i_azi_,i_pol_)
+
+      end if
+
+    end function nuclideangle_calc_f
+
+!===============================================================================
+! find_angle finds the closest angle on the data grid and returns that index
+!===============================================================================
+
+    pure subroutine find_angle(polar, azimuthal, uvw, i_azi, i_pol)
```

Comment:
This seems like a weird file for this subroutine to be in, but I'm not sure where it should ideally go.